sqlmesh/core/plan/evaluator.py

```diff
@@ -88,6 +89,9 @@ def evaluate(
         circuit_breaker: t.Optional[t.Callable[[], bool]] = None,
     ) -> None:
         self._circuit_breaker = circuit_breaker
+        self.snapshot_evaluator = self.snapshot_evaluator.set_correlation_id(
```

Comment:
I don't see us closing this snapshot evaluator in the `finally` block of the plan evaluator

Reply:
That's fine. This means you need to call `.recycle` instead of `close`.